### ISSUE TITLE
MaxPooling2D in Keras fails due to TypeError in backend

### ISSUE THREAD

**Issue by phrz** [plaidml/plaidml#61]
I do not claim to even slightly understand the inner workings of Keras or PlaidML, but I was determined to hunt this issue down. Whenever I tried to add a keras `MaxPooling2D` layer to a `Sequential`, I received the following error:
```
TypeError: can only concatenate tuple (not "list") to tuple
```
Attributed to [this line](https://github.com/plaidml/plaidml/blob/7f3285a4364bb3a79b1082bd28fd52d487019b01/plaidml/keras/backend.py#L2097) in plaidml/keras/backend.py:
```python
outshape = x.shape[:2] + num_out_size
```
Without understanding much of the surrounding code, other than that its purpose was to generate Tile code, I found that `x.shape` resulted in a tuple `(None, 1)`, yet `num_out_size` was an array of integers. After some educated guessing, I changed the line to this:
```python
outshape = list(x.shape[:2]) + num_out_size
```
and no longer received the error. I am running `plaidml 0.1.2`, `keras 2.0.8`, `plaidml-keras 0.1.2`, on Python 3.6.3 installed through Homebrew (`brew install python3`) on a Mac running macOS 10.13. I produced the following example code to reproduce the error.
```python
import plaidml.keras
plaidml.keras.install_backend()

import keras
from keras.models import Sequential
from keras.layers import Reshape, MaxPooling2D, Conv2D

model = Sequential()

model.add(
	Reshape(
		(1, 64, 64), 
    		input_shape=(1, 64*64)
	)
)

model.add(
	Conv2D(
		filters=3,
		kernel_size=(5,5),
		padding='same',
		activation='relu'
	)
)

model.add(
	MaxPooling2D(pool_size=(2,2), data_format='channels_first')
)
```
I cannot guarantee that this is a *minimal* example, it is just a very faithful recreation of the original code that triggered the error, with hardcoded values instead of external variables.

If I am misguided in submitting this issue, which I do not doubt, please point me in the right direction. I just wanted to see if this problem could be fixed through a little tinkering.

**Comment by phrz** [comment]
i would like to **strongly qualify** this by adding that i do **not** suggest that line change as a solution. It was just part of the probing process. in the code this sample comes from, test accuracy dropped to 50%+/-0 (always classifying as one class in a binary problem), so it almost definitely *breaks things*. The only thing I’m relatively certain of is the existence of a typing problem in the backend code.

**Comment by earhart** [comment]
Thanks for the bug report, and the simple repro!  The typing error is definitely real.  This will be fixed in the next release.

**Comment by brianretford** [comment]
This should be fixed at head and in the just released 0.1.3
